### PR TITLE
Use msgpack for error log

### PR DESCRIPTION
**Pull request by kristofferkoch**

### CODE REVIEW

**Comment by kristofferkoch** [comment]
Unit tests for legacy reading is missing.

File: src/components/api.ts [hunk 1]
```diff
@@ -412,6 +413,36 @@ export default class Api {
...
+    } catch (e) {
+      this.getErrorLogMsgPackSupport = false;
+      this.logger.debug(
+        'ErrorLog MessagePack not supported on this device. Using legacy procedure!',
+        'SDK API'
+      );
+      return this.getErrorLogLegacy(timeout);
+    }
+  }
+
```
Comment (brikendr) [inline src/components/api.ts]: Awesome

File: tests/components/api.spec.ts [hunk 2]
```diff
@@ -416,16 +416,26 @@ describe('API', () => {
...
+        sendAndReceiveMsgPackStub.resolves(errorLogDummy);
         const log = await api.getErrorLog(100);
-        expect(fileTransferStub.callCount).to.equals(1);
-        expect(transport.write.firstCall.args[0]).to.equals('error_logger/read');
+        expect(sendAndReceiveMsgPackStub.firstCall.args[1]).to.deep.equals({
+          send: 'error_logger/read_simple', receive: 'error_logger/read_simple_reply'
+        });
         expect(log).to.equals(logStr);
       });
     });
```
Comment (brikendr) [inline tests/components/api.spec.ts]: Plan to add unit tests for the fallback funactionality?

**Comment by brikendr** [comment]
Also could you use `git cz` for the commits?

**Pull request merged**